crates/db_views/src/post_view.rs

```diff
@@ -410,6 +403,10 @@ impl<'a> PostQueryBuilder<'a> {
       query = query.filter(post_read::id.is_null());
     }
 
+    if let Some(languages) = self.languages {
+      query = query.filter(post::language.eq(any(languages)))
+    }
```

Comment:
`in` queries / eq_any ones are terrible for performance, this needs to be a join to a user_discussion_language table.

Reply:
I couldnt find any conclusive info about the performance, but it makes sense if there are big arrays. And if we initialize new users with all languages enabled, the discussion_languages array will have around 180 elements, so thats definitely a lot. Maybe as a workaround, we could set an empty array for discussion_languages in sql, and interpret that as "all languages enabled" (so the filter could be skipped in that case). We could also set a maximum number of enabled languages.

Reply:
You can't have foreign key constraints, indexes, fast lookups and joins, everything SQL was meant to do quickly.